### ISSUE TITLE
Keyword-like forms of logical operators are not supported in C++ requires clause

### ISSUE THREAD

**Issue by MiroPalmu** [doxygen/doxygen#10506]
**Describe the bug**
Keyword-like forms of logical operators `and`, `or` and `not` are not supported in required clause.

For example for following class:
```
template<typename T>
    requires std::is_enum_v<T> and std::unsigned_integral<std::underlying_type_t<T>>
class flags_t {...};
```
"Detailed Description" in html documentation would contain:
```
template<typename T>
requires std::is_enum_v<T>
class idg::sstd::flags::flags_t< T >
```

If one would change `and` -> `&&` missing `std::unsigned_integral<std::underlying_type_t<T>>` would appear in the documentation.

**Expected behavior**
Support for keyword-like forms of logical opeators.

**Screenshots**
![image](https://github.com/doxygen/doxygen/assets/104301888/7934cd62-f01a-451d-9962-87e34e300b83)

**To Reproduce**
[doxygen_bug.tar.gz](https://github.com/doxygen/doxygen/files/13755088/doxygen_bug.tar.gz)

**Version**
Doxygen 1.9.6 on Arch Linux

**Additional context**
I don't understand the implementation enough to say for sure but it seems that the bug lies in doxygen/src/scanner.I:2343:

```
<RequiresClause>"||"|"&&"               { // "requires A || B" or "requires A && B"
                                          yyextra->current->req+=yytext;
                                        }
```

**Comment by albert-github** [comment]
What kind of file is your `Doxyfile` in the `doxygen_bug.tar.gz`?

**Comment by albert-github** [comment]
In your code you have
```
requires std::is_enum_v<T> and std::unsigned_integral<std::underlying_type_t<T>>
```
- from where do you get that the word `and` is supported?
- does did this compile?

As part of the draft C++ 2023 standard I see:
```
requires-clause:
  requires constraint-logical-or-expression

constraint-logical-or-expression:
  constraint-logical-and-expression
  constraint-logical-or-expression || constraint-logical-and-expression

constraint-logical-and-expression:
  primary-expression
  constraint-logical-and-expression && primary-expression
```
so, I think, for `or` you have use `||`, for `and` use `&&` and for `not` use `!`.

**Comment by doxygen** [comment]
@albert-github These are so called "alternative operators", see https://en.cppreference.com/w/cpp/language/operator_alternative

**Comment by albert-github** [comment]
@doxygen  I see (found them also in the standard as paragraph "5.5 Alternative tokens", some horrible sequences like `<%` and `<:` I see there. Maybe we should support the `and`, `or`, `not` as well where now the `&&`, `||` and `!` are supported (in scanner.l and code.l). Probably some others as well.

**Comment by MiroPalmu** [comment]
> What kind of file is your `Doxyfile` in the `doxygen_bug.tar.gz`?

It is just the default generated one. There is also main.cpp which contains test cases for each `and`, `or` and `not`.

**Comment by doxygen** [comment]
> @doxygen I see (found them also in the standard as paragraph "5.5 Alternative tokens", some horrible sequences like `<%` and `<:` I see there. Maybe we should support the `and`, `or`, `not` as well where now the `&&`, `||` and `!` are supported (in scanner.l and code.l). Probably some others as well.

I think we should definitely not support the weird `<%` and `<:` sequences, I've never seen them used, so let's only do the `&&` and `||` and `!` and only at places where they make sense. Note that `&&` is also used for forwarding/move references (see https://stackoverflow.com/a/71285923/784672 for some horrible abuse).

**Comment by doxygen** [comment]
@MiroPalmu Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by MiroPalmu** [comment]
@doxygen Yes, it worked! Thanks for the quick fix :)

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.10.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
